### ISSUE TITLE
Strapi.query doesn't get signedUrls back in contrast to entityService queries

### ISSUE THREAD

**Issue by kevinvugts** [strapi/strapi#18963]
### Required System information

- Node.js version: v18.18.2
- NPM version: 9.8.1
- Strapi version: 4.15.4
- Database: Mysql
- Operating system: Mac OS Ventura
- Is your project Javascript or Typescript: Javascript

### Describe the bug

If an collection type contains a media file and you execute a find request using the strapi.query(uid).find  or strapi.query(uid).findOne methods the signedUrl provided by the private provider isn't being called. This results in a entity you get back containing an url that is not signed. 

In contrary, the entity service does this correctly by calling ["addSignedFileUrlsToEntityService"](https://github.com/strapi/strapi/blob/3301d58a39e2554dbcd016c398aaa62609fe3173/packages/core/upload/server/services/extensions/core/entity-service.js#L5). 

This function is traversing through the entities query result and mutates the file.url to the signedUrl. Which is expected behaviour if you have a private provided implemented.

### Steps to reproduce the behavior

1. Create a collection type with a media attribute (for example: thumbnail)
2. Create a new entity and upload a picture to the thumbnail attribute
3. Use a private provider that implements getSignedUrl() and isPrivate() { return true }
4. Make a custom find controller and use the entityService and confirm that the url is actually signed by the getSignedUrl and that it has a property of isUrlSigned: true. 
5. Do the same as step 4 but this time use the strapi.query(uid).find method. And confirm that this time the url is not signed and the value is coming straight from the database.

### Expected behavior

EntityService and Strapi Query engine are both for different use cases. I am fully aware of this. However, entityService in the end leveraces the queryEngine anyway. In my opinion, the signedUrl should both be implemented on query engine and and entity service level. Currently, the lack of support in the query engine is causing a lot of trouble when the strapi.entityService queries don't provide enough flexibility for certain queries. 

In short, media files urls should be signed if you query for an entity that contains a media file for both queryEngine as entityService (already implemented here).

### Screenshots

Not applicable.

### Code snippets

Relevant code snippets / files related to this issue:

https://github.com/strapi/strapi/blob/3301d58a39e2554dbcd016c398aaa62609fe3173/packages/core/upload/server/services/extensions/core/entity-service.js#

https://github.com/strapi/strapi/blob/63eaab092352f50b83070aa8f7471ed4f95e264f/packages/core/upload/server/services/file.js

https://github.com/strapi/strapi/blob/develop/packages/core/upload/server/services/extensions/utils.js

https://github.com/strapi/strapi/blob/develop/packages/core/utils/src/traverse-entity.ts

### Additional context

Not applicable
